src/components/dockerRegistry/ManageResgistry.tsx

```diff
@@ -0,0 +1,291 @@
+export const CredentialType = {
+    SAME_AS_REGISTRY: 'SAME_AS_REGISTRY',
+    NAME: 'NAME',
+    CUSTOM_CREDENTIAL: 'CUSTOM_CREDENTIAL',
+}
+
+function ManageResgistry({
+    clusterOption,
+    blackList,
+    setBlackList,
+    whiteList,
+    setWhiteList,
+    onEditBlackList,
+    setOnEditBlackList,
+    setOnEditWhiteList,
+    onEditWhiteList,
+    credentialsType,
+    setCredentialType,
+    credentialValue,
+    setCredentialValue,
+    onClickHideManageModal,
+}) {
+    const onClickAlertEditConfirmation = (): void => {
+        if (whiteList.length > 0) {
+            setWhiteList([])
+
+        } else {
+            setBlackList([])
+        }
+        onClickCloseButton()
+    }
+
+    const onClickCloseButton = () => {
+        onClickWhiteListEditHideAlertInfo()
+        onClickBlackListEditHideAlertInfo()
+    }
+
+    const renderActionButton = (): JSX.Element => {
+        return <Close className="cursor" onClick={onClickCloseButton} />
+    }
+
+    const renderAlertMessage = () => {
+        return (
+            <div>
+                If you want to edit this, below selection will not be applicable.
+                <span className="cb-5 cursor ml-4 fw-6" onClick={onClickAlertEditConfirmation}>
+                    Confirm to edit
+                </span>
+            </div>
+        )
+    }
+
+    const renderEditAlert = (): JSX.Element => {
+        return (
+            <InfoColourBar
+                classname="warn"
+                Icon={Warn}
+                message={renderAlertMessage()}
+                iconClass="warning-icon"
+                renderActionButton={renderActionButton}
+            />
+        )
+    }
+
+    const onClickBlackListEditShowAlertInfo = () => {
+        setOnEditBlackList(true)
+    }
+
+    const onClickBlackListEditHideAlertInfo = () => {
+        setOnEditBlackList(false)
+    }
+
+    const onClickWhiteListEditShowAlertInfo = (): void => {
+        !onEditWhiteList && setOnEditWhiteList(true)
+    }
+
+    const onClickWhiteListEditHideAlertInfo = () => {
+      onEditWhiteList && setOnEditWhiteList(false)
+    }
+
+    const renderNoSelectionview = (onEdit) => {
+        return (
+            <div className="flex en-2 bw-1 bcn-1 pr-20">
+                <input
+                    tabIndex={1}
+                    placeholder="None"
+                    className="form__input form__input__none bcn-1"
+                    name="blacklist-none"
+                    disabled={true}
+                    autoFocus
+                    autoComplete="off"
+                />
+                <div onClick={() => onEdit()}>
+                    <Edit className="cursor pr-20" />
+                </div>
+            </div>
+        )
+    }
+
+    console.log(onEditWhiteList)
+    const renderIgnoredCluster = (): JSX.Element => {
+        if (whiteList.length > 0) {
+            return renderNoSelectionview(onClickWhiteListEditShowAlertInfo)
+        } else if (whiteList.length === 0) {
```

Comment:
we can replace else of with else

Reply:
Now we cant. as we adding condition on else if statment